daemon/daemon.go

```diff
@@ -279,7 +279,7 @@ func (d *Daemon) QueueEndpointBuild(epID uint64) func() {
 	}
 
 	// Acquire succeeded, but the context was canceled after?
-	if ctx.Err() == context.Canceled {
+	if ctx.Err() != nil {
```

Comment:
I might be missing it but, in older code, how would `ctx.Err()` ever be non-nil? It is created with `context.WithCancel(context.Background())`

Reply:
Fixed this by creating a new context with the `ctx` parameter from `QueueEndpointBuild` as the parent context.